src/neo/SmartContract/ApplicationEngine.Storage.cs

```diff
@@ -101,9 +101,9 @@ private void PutExInternal(StorageContext context, byte[] key, byte[] value, Sto
             {
                 if (item.IsConstant) throw new InvalidOperationException();
                 if (value.Length <= item.Value.Length)
-                    newDataSize = 1;
+                    newDataSize = (1 + value.Length / 4);
```

Comment:
If he remove information, why we should charge him `value/4` ?

Reply:
As the full state node will retain historical data, we need to use the charging mechanism to prevent the full state node from being attacked.